src/checkpoint/graphql/resolvers.ts

```diff
@@ -9,6 +9,14 @@ export interface ResolverContext {
   mysql: AsyncMySqlPool;
 }
 
+const getTableName = (fieldName: string): string => {
+  if (fieldName.startsWith('_')) {
+    return fieldName;
+  }
+
+  return `${fieldName}s`;
```

Comment:
I'm thinking that instead just using `s` for plurals, we start using a library like [pluralize](https://www.npmjs.com/package/pluralize).

Because right now the queries for the _metadata table are `_metadata` and `_metadatas`, which is kind of weird but works for now, but I can imagine if a user defined an entity like `Box`, we will be generating a `boxs` queries as opposed to a `boxes` query. 

I'll create an issue for this and we can pick it up later.

Reply:
Actually thinking more about this now and the `_checkpoint`/`_checkpoints` queries might be a problem because with the check in the above function the suffix `s` will not be used when querying the table. 

But if we were using a pluralize function, then `_checkpoints` will always returns as `_checkpoints`, also the singular function will always return to singular value.

Reply:
This is not a big concern, I believe TheGraph just add a "s" for plural even if you use "box". We will never have a perfect plural anyway, some words are always written in singular.